socket/sandboxes/heka/output/heka_tcp.lua

```diff
@@ -59,16 +59,16 @@ end
 
 local client, err = create_client()
 
-local function send_message(msg, i)
-    local len, err, i = client:send(msg, i)
-    if not len then
-        if err == "timeout" or err == "closed" then
+local function send_message(msg)
+    local i, len, err = 0
+    repeat
+        len, err, i = client:send(msg, i + 1)
+        if not len then
```

Comment:
Looking at https://github.com/diegonehab/luasocket/blob/master/src/buffer.c#L75 any error (even a recoverable one) would have a nil len so we would never retry any partial result